### PR TITLE
Update the project to use docker compose

### PR DESCRIPTION
**Pull request by giteshk**
Changes added:
- Use docker compose to run multiple containers
- Introduce MariaDB as the DB
- Update the docker file to use standard php-fpm image from docker hub.
- Setup a folder structure so customers can use it to run their own images in production.
- Image does not build with the site installed anymore.
- Startup file will run drush installation if `AUTO_INSTALL_PORTAL` environment variable is set to true.

Site is available at http://localhost:8080

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -14,14 +14,4 @@
...
-
-docker run --name apigee-d8 -p 8080:80 -d \
-	-e APIGEE_EDGE_AUTH_TYPE=basic \
-	-e APIGEE_EDGE_ORGANIZATION=$APIGEE_ORG \
-	-e APIGEE_EDGE_USERNAME=$APIGEE_USER \
-	-e APIGEE_EDGE_PASSWORD=$APIGEE_PASS \
-	-e APIGEE_EDGE_ENDPOINT=$APIGEE_MGMT \
-	ghcr.io/apigee/docker-apigee-drupal-kickstart:latest
\ No newline at end of file
+docker compose -f docker-compose-demo.yml up
```
[the remark this answers is not in the document]
Comment (giteshk) [reply run.sh:17]: done

File: docker-compose-demo.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+version: '3'
+services:
+  apigee-kickstart:
+    image: ghcr.io/apigee/docker-apigee-drupal-kickstart:latest
```
[the remark this answers is not in the document]
Comment (giteshk) [reply docker-compose-demo.yml:18]: agreed. I managed to find a way to make this work with one docker-compose.yml file

File: docker-compose-demo.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    links:
+    - apigee-kickstart-database
+    depends_on:
+      apigee-kickstart-database:
+        condition: service_healthy
+    volumes:
+    - drupal-public-files:/app/code/web/sites/default/files
+    - drupal-private-files:/app/code/web/sites/default/private
+  apigee-kickstart-database:
+    image: mariadb:latest
```
[the remark this answers is not in the document]
Comment (giteshk) [reply docker-compose-demo.yml]: Agreed. To avoid any unnecessary surprises

**Comment by giteshk** [comment]
@danistrebel Thanks a lot for the review. Appreciate you taking the time to test and provide feedback.

**Review by danistrebel — APPROVED** [review]

**Pull request merged**